File: frontend/src/types/clothing.ts
export type ClothingCategory = "topwear" | "bottomwear" | "footwear" | "accessory" | "outerwear";

export type ClothingPattern =
  | "solid" | "striped" | "checked" | "printed" | "floral" | "graphic" | "textured";

export type ClothingStyle =
  | "casual" | "formal" | "business_casual" | "smart_casual" | "party" | "sports" | "ethnic";

export type Season = "summer" | "winter" | "rainy" | "all_season";

export type Gender = "male" | "female" | "unisex";

export interface ClothingItem {
  id: number;
  user_id: number;
  image_path: string;
  category: ClothingCategory;
  subcategory: string;
  primary_color: string;
  secondary_color: string | null;
  pattern: ClothingPattern;
  style: ClothingStyle;
  season: Season;
  gender: Gender;
  tags: string[];
  is_active: boolean;
  wear_count: number;
  last_worn: string | null;
  created_at: string;
}

export interface ClothingItemSummary {
  id: number;
  category: ClothingCategory;
  subcategory: string;
  primary_color: string;
  style: ClothingStyle;
  image_path: string;
}

export interface ClothingItemUpdate {
  category?: ClothingCategory;
  subcategory?: string;
  primary_color?: string;
  secondary_color?: string | null;
  pattern?: ClothingPattern;
  style?: ClothingStyle;
  season?: Season;
  gender?: Gender;
  tags?: string[];
  is_active?: boolean;
}

export interface WardrobeFilters {
  category?: ClothingCategory;
  style?: ClothingStyle;
  season?: Season;
  color?: string;
  pattern?: ClothingPattern;
}

export const CATEGORIES: ClothingCategory[] = [
  "topwear", "bottomwear", "footwear", "outerwear", "accessory",
];

export const STYLES: ClothingStyle[] = [
  "casual", "formal", "business_casual", "smart_casual", "party", "sports", "ethnic",
];

export const SEASONS: Season[] = ["summer", "winter", "rainy", "all_season"];

export const PATTERNS: ClothingPattern[] = [
  "solid", "striped", "checked", "printed", "floral", "graphic", "textured",
];

File: backend/app/schemas/clothing.py
from __future__ import annotations
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


# ── Enums as literals ────────────────────────────────────────────────────────

CATEGORIES = ["topwear", "bottomwear", "footwear", "accessory", "outerwear"]
PATTERNS = ["solid", "striped", "checked", "printed", "floral", "graphic", "textured"]
STYLES = ["casual", "formal", "business_casual", "smart_casual", "party", "sports", "ethnic"]
SEASONS = ["summer", "winter", "rainy", "all_season"]
GENDERS = ["male", "female", "unisex"]


# ── Request Schemas ───────────────────────────────────────────────────────────

class ClothingItemCreate(BaseModel):
    category: str = Field(..., description="topwear | bottomwear | footwear | accessory | outerwear")
    subcategory: str = Field(..., description="e.g. t-shirt, jeans, oxford shirt")
    primary_color: str
    secondary_color: Optional[str] = None
    pattern: str = Field(default="solid")
    style: str = Field(..., description="casual | formal | business_casual | smart_casual | party")
    season: str = Field(default="all_season")
    gender: str = Field(default="unisex")
    tags: List[str] = Field(default_factory=list)


class ClothingItemUpdate(BaseModel):
    """All fields optional — patch semantics."""
    category: Optional[str] = None
    subcategory: Optional[str] = None
    primary_color: Optional[str] = None
    secondary_color: Optional[str] = None
    pattern: Optional[str] = None
    style: Optional[str] = None
    season: Optional[str] = None
    gender: Optional[str] = None
    tags: Optional[List[str]] = None
    is_active: Optional[bool] = None


# ── Response Schemas ──────────────────────────────────────────────────────────

class ClothingItemResponse(BaseModel):
    id: int
    user_id: int
    image_path: str
    category: str
    subcategory: str
    primary_color: str
    secondary_color: Optional[str]
    pattern: str
    style: str
    season: str
    gender: str
    tags: List[str]
    is_active: bool
    wear_count: int
    last_worn: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True


class ClothingItemSummary(BaseModel):
    """Lightweight version used inside outfit responses."""
    id: int
    category: str
    subcategory: str
    primary_color: str
    style: str
    image_path: str

    class Config:
        from_attributes = True


class WardrobeFilterParams(BaseModel):
    category: Optional[str] = None
    style: Optional[str] = None
    season: Optional[str] = None
    color: Optional[str] = None
    pattern: Optional[str] = None
    is_active: bool = True
